app/build.gradle.kts

```diff
@@ -0,0 +1,91 @@
+apply(from = rootProject.file("versions.gradle.kts"))
+
+plugins {
+    id("com.android.application")
+    id("org.jetbrains.kotlin.android")
+    id("kotlin-kapt")
+    id("com.google.dagger.hilt.android") version "2.48"
+}
+
+println("compose_version in build script: ${extra["compose_version"]}")
+
+android {
+    namespace = "com.example.chessclockk"
+    compileSdk = 34
+
+    defaultConfig {
+        applicationId = "com.example.chessclockk"
+        minSdk = 26
+        targetSdk = 34
+        versionCode = 1
+        versionName = "1.0"
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        vectorDrawables {
+            useSupportLibrary = true
+        }
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+            proguardFiles(
+                getDefaultProguardFile("proguard-android-optimize.txt"),
+                "proguard-rules.pro"
+            )
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    buildFeatures {
+        compose = true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion = "1.5.1"
+    }
+    packaging {
+        resources {
+            excludes += "/META-INF/{AL2.0,LGPL2.1}"
+        }
+    }
+}
+
+tasks.withType<Test>{
+    useJUnitPlatform()
+}
+
+hilt {
+    enableAggregatingTask = false
+}
+
+dependencies {
+    implementation("com.google.dagger:hilt-android:2.48")
```

Comment:
Teraz wersje dodaje się w TOMLu https://developer.android.com/build/migrate-to-catalogs
Wtedy są referencje na numer wersji. To głównie przez modularyzacje, wtedy warto mieć jedno miejsce z wersjami